src/reducers/vms.js

```diff
@@ -28,66 +27,58 @@ import {
 import { actionReducer, removeMissingItems } from './utils'
 import { sortFields } from '_/utils'
 
-const EMPTY_MAP = Immutable.fromJS({})
-const EMPTY_ARRAY = Immutable.fromJS([])
-
 const initialState = Immutable.fromJS({
   vms: {},
   pools: {},
-  filters: EMPTY_MAP,
+  filters: {},
   sort: { ...sortFields[0], isAsc: true },
 
-  page: 1,
-
   missedVms: Immutable.Set(),
 
-  /**
-   * true ~ we need to fetch further vms and pools
-   * false ~ all visible entities already fetched
-   */
-  notAllPagesLoaded: true,
+  vmsPage: 0,
```

Comment:
consider storing also `pageCount` in the store

Reply:
This is the page count for VMs.

`poolsPage` is the page count for Pools.

Reply:
Sorry for not being clear enough - `pageCount` is just bad name. My proposal was about adding a new field that would store the number of items per page so `pageSize` is the correct name. Currently we are using value from configuration i.e..
```
count: vmsPage * AppConfiguration.pageLimit,
```
This is a good candidate to be added to user settings (at some point)

Reply:
Ah ok - that would be better placed in the `config` reducers. So, the
 ```jsx
  count: vmsPage * AppConfiguration.pageLimit
```
would just turn into something like:
```jsx
  count: vmcPage * yield select(state => state.config.get('fetchPageSize'))
```

but we can cross that bridge when we get to it.